library/lua/gui/widgets.lua

```diff
@@ -10,12 +10,17 @@ local utils = require('utils')
 local getval = utils.getval
 local to_pen = dfhack.pen.parse
 
+---@param view gui.View
+---@param vis boolean
 local function show_view(view,vis)
     if view then
         view.visible = vis
     end
 end
 
+---@param tab table
```

Comment:
`tab? table<T>`

Reply:
As mentioned on Discord, I think this doesn't work. Full example would be something like:

```lua
---@generic T
---@param tab? table<T>
---@param idx integer
---@return T
local function map_opttab(tab,idx)
    if tab then
        return tab[idx]
    else
        return idx
    end
end

---@type string[]
local foo
local bar = map_opttab(foo, 0)```

Where `bar` remains `unknown`